inspirehep/modules/citations/receivers.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2015 CERN.
+#
+# INSPIRE is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+
+from invenio_records.signals import after_record_insert, after_record_update
+
+from .tasks import update_citation_count_for_records
+
+
+@after_record_insert.connect
+def catch_citations_insert(sender, *args, **kwargs):
+
+    records_to_notify = []
+    if 'references' in sender:
+            records_to_notify = [ref['recid'] for ref in sender['references'] if ref.get('recid')]
+
+    if records_to_notify:
+        # countdown added as it takes a while before an indexed record can be accessed from es
+        update_citation_count_for_records.apply_async(args=[records_to_notify], countdown=3)
+
+
+@after_record_update.connect
+def catch_citations_update(sender, *args, **kwargs):
+
+    refs_before_update = []
+    if sender.init_json:
+        if 'references' in sender.init_json:
+            refs_before_update = [ref['recid'] for ref in sender.init_json['references'] if ref.get('recid')]
+
+    refs_after_update = []
+    if sender.init_json:
+        if 'references' in sender:
+            refs_after_update = [ref['recid'] for ref in sender['references'] if ref.get('recid')]
+
+    refs_diff = filter(lambda x: x not in refs_before_update, refs_after_update) + \
```

Comment:
Infact, you make `refs_before_update` and `refs_after_update` to be `set`s, then `refs_diff = refs_before_update ^ refs_after_update`.